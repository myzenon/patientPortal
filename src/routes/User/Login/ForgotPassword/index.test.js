import React from 'react'
import { ForgotPassword } from './index'
import ReactGA from 'react-ga';

   // This should be part of your setup
   ReactGA.initialize('UA-121783543-1', {
       'debug': true
     });;

jest.mock("react-ga")

test('renders correctly', () => {
    const wrapper = render (
        <ForgotPassword />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
})

test('email validation on user unfocus email entry box', () => {
    const wrapper = mount (
        <ForgotPassword />
    )
    wrapper.find('input[name="email"]').simulate('change', {
        target: {
            name: 'email',
            value: 'abcde'
        }
    })
    wrapper.find('input[name="email"]').simulate('blur')
    expect(toJson(wrapper.render())).toMatchSnapshot()
})

test('clicks NEXT | api forgot password call | show Password Reset Email Has Been Sent dialog box', (done) => {
    const mockEmail = 'test@test.com'
    const props = {
        callAPIGateway: (name, get, post) => {
            expect(name).toBe('usersForgotPasswordPost')
            expect(post).toEqual({
                creds: {
                    user_name: mockEmail
                }
            })
            return Promise.resolve()
        }
    }
    const wrapper = mount (
        <ForgotPassword {...props} />
    )
    wrapper.find('input[name="email"]').simulate('change', {
        target: {
            name: 'email',
            value: mockEmail
        }
    })
    wrapper.find('input[name="email"]').simulate('blur')
    wrapper.find('button[type="submit"]').simulate('submit')
    setTimeout(() => {
        expect(toJson(wrapper.render())).toMatchSnapshot()
        done()
    }, 0)
})