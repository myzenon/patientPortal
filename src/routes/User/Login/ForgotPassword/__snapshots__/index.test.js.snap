// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clicks NEXT | api forgot password call | show Password Reset Email Has Been Sent dialog box 1`] = `
<div>
  <div
    class="wrapper"
  >
    <div
      class="header"
    >
      <div
        class="headerText"
      >
        Password reset email has been sent!
      </div>
      <div
        class="headerClose"
      >
        X
      </div>
    </div>
    <div
      class="content"
    >
      <p>
        We have sent a password reset email to 
        <strong>
          test@test.com
        </strong>
        .
      </p>
      <p>
        Please check your email in a few minutes. If you haven't received it, please check your spam folder.
      </p>
      <p>
        The email will be from 
        <strong />
        .
      </p>
      <p>
        If you do not receive an email, please call Arine at  
        <strong>
           
        </strong>
        .
      </p>
    </div>
    <div
      class="action"
    >
      <button
        class="btn btn-secondary"
        type="button"
      >
        Done
      </button>
    </div>
  </div>
</div>
`;

exports[`email validation on user unfocus email entry box 1`] = `
<div>
  <form
    class=""
  >
    <div
      class="wrapper"
    >
      <div
        class="header"
      >
        <div
          class="headerText"
        >
          Password Reset
        </div>
        <div
          class="headerClose"
        >
          X
        </div>
      </div>
      <div
        class="content"
      >
        <p>
          Enter your email address. If you are registered, we will send you an email with instructions on how to reset your password.
        </p>
        <div>
          <input
            class="inputTextError"
            name="email"
            placeholder="Enter your email address here"
            type="email"
          />
          <div
            class="inputMessageError"
          >
            Please enter a valid email address.
          </div>
        </div>
      </div>
      <div
        class="action"
      >
        <button
          class="btn btn-secondary disabled"
          disabled=""
          type="submit"
        >
          Send Email
        </button>
      </div>
    </div>
  </form>
</div>
`;

exports[`renders correctly 1`] = `
<div>
  <form
    class=""
  >
    <div
      class="wrapper"
    >
      <div
        class="header"
      >
        <div
          class="headerText"
        >
          Password Reset
        </div>
        <div
          class="headerClose"
        >
          X
        </div>
      </div>
      <div
        class="content"
      >
        <p>
          Enter your email address. If you are registered, we will send you an email with instructions on how to reset your password.
        </p>
        <div>
          <input
            class=""
            name="email"
            placeholder="Enter your email address here"
            type="email"
          />
        </div>
      </div>
      <div
        class="action"
      >
        <button
          class="btn btn-secondary"
          type="submit"
        >
          Send Email
        </button>
      </div>
    </div>
  </form>
</div>
`;
